src/Features/Core/Portable/ExtractMethod/MethodExtractor.VariableSymbol.cs

```diff
@@ -21,294 +17,172 @@ internal abstract partial class AbstractExtractMethodService<
 {
     internal abstract partial class MethodExtractor
     {
-        /// <summary>
-        /// temporary symbol until we have a symbol that can hold onto both local and parameter symbol
-        /// </summary>
-        protected abstract class VariableSymbol
+        protected abstract class VariableSymbol : IComparable<VariableSymbol>
```

Comment:
cleaned up this entire inheritance hierarchy.  shared a lot more code.  made some parts more strongly typed.